xds/src/main/java/io/grpc/xds/OrcaOobService.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.BindableService;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implements the service/APIs for Out-of-Band metrics reporting, only for utilization metrics.
+ * Register the returned service {@link #getService()} to the server, then a client can request
+ * for periodic load reports. A user should use the public set-APIs to update the server machine's
+ * utilization metrics data.
+ */
+@io.grpc.ExperimentalApi("https://github.com/grpc/grpc-java/issues/9006")
+public final class OrcaOobService {
+  /**
+   * Empty or invalid (non-positive) minInterval config in will be treated to this default value.
+   */
+  public static final long DEFAULT_MIN_REPORT_INTERVAL_NANOS = TimeUnit.SECONDS.toNanos(30);
+
+  private final OrcaServiceImpl orcaService;
+
+  /**
+   * Construct an OOB metrics reporting service.
+   *
+   * @param minInterval configures the minimum metrics reporting interval for the service. Bad
+   *        configuration (non-positive) will be overridden to service default (30s).
+   *        Minimum metrics reporting interval means, if the setting in the client's
+   *        request is invalid (non-positive) or below this value, they will be treated
+   *        as this value.
+   */
+  public OrcaOobService(long minInterval, TimeUnit timeUnit,
+                        ScheduledExecutorService timeService) {
+    this.orcaService = new OrcaServiceImpl(minInterval > 0 ? timeUnit.toNanos(minInterval)
+        : DEFAULT_MIN_REPORT_INTERVAL_NANOS, checkNotNull(timeService));
+  }
+
+  public OrcaOobService(ScheduledExecutorService timeService) {
```

Comment:
FWIW, I think we may end up passing these arguments to `getService()` (renamed as createService). But we'll discuss later.